File: HomeWorks/CPP/Analysis.cpp
#include<iostream>
using namespace std;

bool IsPrime(int num)
{
    bool isprime;
    if(num<=1)
        return false;
    else if(num==2)
        return true;
    else
    {
        isprime = true;
        for(int i=3s; i<num; i++)
        {
            if(num%i==0)
            {
                isprime=false;
                break;
            }
        }
        if(isprime)
            return true;
        else
            return false;
    }
}

int main()
{
    int num;
    cout<<"enter a number: ";
    cin>>num;

    int PowCnt;
    int i=2;
    while(num!=1)
    {
        PowCnt=0;
        if(IsPrime(i))
        {
            if(num%i==0)
            {
                while(num%i==0)
                {
                    num = (num/i);
                    PowCnt++;
                }
                cout<<i<<"^"<<PowCnt<<" * ";
            }
        }

        i++;
    }
    cout<<"1\n\n";

    return 0;
}
//MadMad_59

File: homeworks/py/analysis.py
def IsPrime(num):
    if(num<=1):
        return False
    elif(num==2):
        return True
    else:
        i=2
        prime = True
        while(i<num):
            if(num%i==0):
                prime = False
                break
            i+=1
        if(prime):
            return True
        else:
            return False


num = int(input("n : "))
i=2
while(num!=1):
    PowCnt = 0
    if(IsPrime(i)):
        if(num%i==0):
            while(num%i==0):
                num/=i
                PowCnt+=1
            print(i,"^",PowCnt," * ",end=" ")    
    i+=1
print(1)
#MadMad_32